### PR TITLE
[OpenVINO Backend] Implements segment_sum and segment_max for OpenVINO backend

### PR DESCRIPTION
**Pull request by goyaladitya05**
This PR implements `keras.ops.segment_sum` and `keras.ops.segment_max` for the OpenVINO backend. 

**Implementation Details:**
- Both operations are implemented using OpenVINO's `scatter_nd_update` with `reduction="sum"` and `reduction="max"`. 
- To handle `segment_ids == -1` (which denotes segments to ignore) without triggering dynamic boolean mask overhead, we map `-1` indices to a dummy `num_segments` index at the end of the buffer. The computation buffer is padded by `+1` along the segment dimension, and we simply `slice` off the final row before returning. This accumulates and discards the ignored values in a statically predictable shape.
- Removed `SegmentSumTest` and `SegmentMaxTest` from [excluded_concrete_tests.txt](cci:7://file:///d:/codes/open-source/keras/keras/src/backend/openvino/excluded_concrete_tests.txt:0:0-0:0).

**Testing:**
- Verified `pytest keras/src/ops/math_test.py -k "test_segment_sum_call or test_segment_max_call"` passes locally.
- Verified backend functionality using `pytest keras/src/backend/openvino/`.

Closes: openvinotoolkit/openvino/issues/34329

### CODE REVIEW

File: keras/src/backend/openvino/math.py [hunk 1]
```diff
@@ -10,16 +10,94 @@
...
+        ).output(0)
+    else:
+        buffer_shape = ov_opset.unsqueeze(
+            num_segments_plus_1, ov_opset.constant(0, Type.i32)
+        ).output(0)
+
+    if reduction_method == "max":
+        init_val = np.array(-np.inf, dtype=np.float32)
+    else:
+        init_val = 0
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/backend/openvino/math.py:79]: fixed.

**Review by hertschuh — APPROVED** [review]
Thanks for implementing these!
